crates/harp/src/object.rs

```diff
@@ -248,6 +248,10 @@ pub fn r_dbl_begin(x: SEXP) -> *mut f64 {
     unsafe { REAL(x) }
 }
 
+pub unsafe fn chr_cbegin(x: SEXP) -> *const SEXP {
+    libr::DATAPTR_RO(x) as *const SEXP
+}
```

Comment:
Everything else here is safe. Do you want to eventually make these `unsafe` again? If so I'd propose keeping it safe for consistency for now and then we move away from it (but to me this feels like it should be safe)

Reply:
hmm I think it's unsafe because it's taking an untyped `SEXP` and it doesn't return any `Result`.

We might argue that it's fine because it throws R errors if the type isn't right (but note that it's an implementation detail that it does so instead of crashing like other accessors like `CAR()`).

But I think the criterion for a safe function is that it guarantees a crash or an R error can't happen (either by dynamic checks feeding the `Err` path of the `Result` or by taking typed inputs with strong guarantees such that errors or crashes can't happen).